File: web/src/api/bookingsApi.js
const API_BASE_URL = import.meta.env.VITE_API_BASE_URL

async function handleResponse(response) {
  if (!response.ok) {
    const errorBody = await response.json().catch(() => ({}))
    throw new Error(errorBody.message || 'Something went wrong.')
  }
  return response.json()
}

export async function createBooking({ userId, bookingDate, items }) {
  const response = await fetch(`${API_BASE_URL}/api/bookings`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ userId, bookingDate, items }),
  })
  return handleResponse(response)
}

export async function getAllBookings() {
  const response = await fetch(`${API_BASE_URL}/api/bookings`)
  return handleResponse(response)
}

export async function getBookingsByStatus(status) {
  const response = await fetch(`${API_BASE_URL}/api/bookings/status/${status}`)
  return handleResponse(response)
}

export async function getBookingsByUser(userId) {
  const response = await fetch(`${API_BASE_URL}/api/bookings/user/${userId}`)
  return handleResponse(response)
}

export async function getBookingById(id) {
  const response = await fetch(`${API_BASE_URL}/api/bookings/${id}`)
  return handleResponse(response)
}

export async function updateBookingStatus(id, { status, adminNotes }) {
  const response = await fetch(`${API_BASE_URL}/api/bookings/${id}/status`, {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ status, adminNotes }),
  })
  return handleResponse(response)
}

export async function cancelBookingByUser(id, userId) {
  const response = await fetch(
    `${API_BASE_URL}/api/bookings/${id}/cancel?userId=${encodeURIComponent(userId)}`,
    { method: 'PATCH' }
  )
  return handleResponse(response)
}

File: application/app/src/main/java/edu/cit/ballener/lakbayayos/data/api/BookingsApi.kt
package edu.cit.ballener.lakbayayos.data.api

import edu.cit.ballener.lakbayayos.data.model.Booking
import edu.cit.ballener.lakbayayos.data.model.BookingRequest
import edu.cit.ballener.lakbayayos.data.model.BookingStatusUpdateRequest
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface BookingsApi {

    // Used by customers
    @POST("/api/bookings")
    suspend fun createBooking(@Body request: BookingRequest): Booking

    @GET("/api/bookings/user/{userId}")
    suspend fun getBookingsByUser(@Path("userId") userId: String): List<Booking>

    @PATCH("/api/bookings/{id}/cancel")
    suspend fun cancelBookingByUser(@Path("id") id: Long, @Query("userId") userId: String): Booking

    // Used by admin
    @GET("/api/bookings")
    suspend fun getAllBookings(): List<Booking>

    @GET("/api/bookings/status/{status}")
    suspend fun getBookingsByStatus(@Path("status") status: String): List<Booking>

    @PATCH("/api/bookings/{id}/status")
    suspend fun updateBookingStatus(@Path("id") id: Long, @Body request: BookingStatusUpdateRequest): Booking
}
